types/collections.go

```diff
@@ -135,6 +141,24 @@ func LengthPrefixedAddressKey[T addressUnion](keyCodec collcodec.KeyCodec[T]) co
 	}
 }
 
+// Deprecated: lengthPrefixedBytesKey is a special key codec used to retain state backwards compatibility
+// when a bytes key is used as an index key.
+type lengthPrefixedBytesKey struct {
+	collcodec.KeyCodec[[]byte]
+}
+
+func (g lengthPrefixedBytesKey) Encode(buffer, key []byte) (int, error) {
+	return g.EncodeNonTerminal(buffer, key)
```

Comment:
Should we make sure the buffer is large enough to store the length prefix?
```suggestion
	return g.EncodeNonTerminal(buffer[:len(key)], key)
```

Reply:
not necessary. We are checking that in usage of bytesKey https://github.com/cosmos/cosmos-sdk/blob/main/collections/codec/bytes.go#L49